server/service/service_osquery.go

```diff
@@ -774,15 +775,20 @@ func (svc *Service) SubmitDistributedQueryResults(
 	}
 
 	if len(labelResults) > 0 {
-		if ac.ServerSettings.DeferredSaveHost {
-			if err := svc.ds.RecordLabelQueryExecutions(ctx, &host, labelResults, svc.clock.Now(), true); err != nil {
-				logging.WithErr(ctx, err)
-			}
-		} else {
-			if err := svc.task.RecordLabelQueryExecutions(ctx, &host, labelResults, svc.clock.Now()); err != nil {
-				logging.WithErr(ctx, err)
-			}
+		// TODO(mna): should async host processing overrule deferred save host? It
+		// is strictly "more async". We could pass the deferred save host setting to
+		// task.RecordLabelQueryExecutions and let it dispatch appropriately (to async
+		// or datastore) - I've applied this approach below.
```

Comment:
I think this is the right approach  ^ , if everyone agrees on this I'll clean up the commented code here.

Reply:
Hm, I think so... they operate at different levels. We might want to just move the whole save host to this redis async method though.